src/frontend/web_application/src/scenes/UserProfile/presenter.jsx

```diff
@@ -28,39 +31,63 @@ class UserProfile extends Component {
     this.props.requestUser();
   }
 
-  handleUserChange = ({ contact, original }) => {
-    this.props.updateContact({ contact, original }).then(() => this.props.requestUser());
+  handleSubmit = async (ev) => {
+    const { handleSubmit } = this.props;
+    await handleSubmit(ev)
+    .then(this.handleSuccess, this.handleError);
+  }
+
+  handleSuccess = async () => {
+    await this.props.requestUser();
+
+    return this.toggleEditMode();
+  }
+
+  handleError = () => {
+    const { i18n, notifyError } = this.props;
+    notifyError({ message: i18n._('contact.feedback.unable_to_save') });
   }
 
   toggleEditMode = () => {
     this.setState({ editMode: !this.state.editMode });
   }
 
+  renderForm = () => {
+    const { i18n, pristine, submitting } = this.props;
+
+    return (
+      <form method="post" name="user-profile" onSubmit={this.handleSubmit}>
+        <ProfileForm editMode={this.state.editMode} />
+        {!this.state.editMode ? (
+          <Button onClick={this.toggleEditMode} shape="plain">
+            {i18n._('user.action.edit_profile')}</Button>
```

Comment:
it's better to use new `<Trans id="user.action.edit_profile">Edit</Trans>` component because it provides a default label for missing translations. (German translations is not 100% translated for now)

Reply:
nevertheless, It might be possible to pass a default translation to i18n._

Reply:
I found it: 

```
{i18n._('user.action.edit_profile', { defaults: 'Edit' })}
```

Reply:
ok thank you